src/api/firebase.ts

```diff
@@ -253,9 +254,41 @@ export async function addItem(
 export async function updateItem(listPath: string, item: ListItem) {
 	const itemDocRef = doc(db, listPath, "items", item.id);
 
-	const updates: Pick<ListItem, "totalPurchases" | "dateLastPurchased"> = {
+	const lastUpdatedDate = item.dateLastPurchased
+		? item.dateLastPurchased
+		: item.dateCreated;
+
+	// Last estimated date of next purchase, or previous dateNextPurchased in whole number
+	const previousEstimate = getDaysBetweenDates(
+		lastUpdatedDate.toDate(),
+		item.dateNextPurchased.toDate(),
+	);
+
+	const daysSinceLastPurchased = getDaysBetweenDates(
+		new Date(),
+		lastUpdatedDate.toDate(),
+	);
+
+	const newDateNextPurchased = getFutureDate(
+		calculateEstimate(
+			previousEstimate,
+			daysSinceLastPurchased,
+			item.totalPurchases,
+		),
+	);
+
+	console.log("item date last purchased:", item.dateLastPurchased?.toDate());
+	console.log("previous estimate:", previousEstimate);
+	console.log("Original next purchase date:", item.dateNextPurchased.toDate());
+	console.log("New Next Purchase Date!", newDateNextPurchased);
```

Comment:
Don't forget to remove your console logs when submitting your PRs! If you really want you can explain the logic in a comment or somewhere in the read me on the utility of the changes.